Mathlib/Algebra/Group/TypeTags.lean

```diff
@@ -0,0 +1,465 @@
+  { Multiplicative.inv with inv_inv := @neg_neg α _ }
+
+instance Additive.subNegMonoid [DivInvMonoid α] : SubNegMonoid (Additive α) :=
+  { Additive.neg, Additive.sub, Additive.addMonoid with
+    sub_eq_add_neg := @div_eq_mul_inv α _
+    zsmul := @DivInvMonoid.zpow α _
+    zsmul_zero := @DivInvMonoid.zpow_zero α _
+    zsmul_succ := @DivInvMonoid.zpow_succ α _
+    zsmul_neg := @DivInvMonoid.zpow_neg α _ }
+
+instance Multiplicative.divInvMonoid [SubNegMonoid α] : DivInvMonoid (Multiplicative α) :=
+  { Multiplicative.inv, Multiplicative.div, Multiplicative.monoid with
+    div_eq_mul_inv := @sub_eq_add_neg α _
+    zpow := @SubNegMonoid.zsmul α _
+    zpow_zero := @SubNegMonoid.zsmul_zero α _
+    zpow_succ := @SubNegMonoid.zsmul_succ α _
+    zpow_neg := @SubNegMonoid.zsmul_neg α _ }
+
+instance Additive.subtractionMonoid [DivisionMonoid α] : SubtractionMonoid (Additive α) :=
+  { Additive.subNegMonoid, Additive.involutiveNeg with
+    neg_add_rev := @mul_inv_rev α _
+    neg_eq_of_add := @inv_eq_of_mul_eq_one_right α _ }
+
+instance Multiplicative.divisionMonoid [SubtractionMonoid α] : DivisionMonoid (Multiplicative α) :=
+  { Multiplicative.divInvMonoid, Multiplicative.involutiveInv with
+    mul_inv_rev := @neg_add_rev α _
+    inv_eq_of_mul := @neg_eq_of_add_eq_zero_right α _ }
+
+instance Additive.subtractionCommMonoid [DivisionCommMonoid α] :
+    SubtractionCommMonoid (Additive α) :=
+  { Additive.subtractionMonoid, Additive.addCommSemigroup with }
+
+instance Multiplicative.divisionCommMonoid [SubtractionCommMonoid α] :
+    DivisionCommMonoid (Multiplicative α) :=
+  { Multiplicative.divisionMonoid, Multiplicative.commSemigroup with }
+
+instance Additive.addGroup [Group α] : AddGroup (Additive α) :=
+  { Additive.subNegMonoid with add_left_neg := @mul_left_inv α _ }
+
+instance Multiplicative.group [AddGroup α] : Group (Multiplicative α) :=
+  { Multiplicative.divInvMonoid with mul_left_inv := @add_left_neg α _ }
+
+instance [CommGroup α] : AddCommGroup (Additive α) :=
+  { Additive.addGroup, Additive.addCommMonoid with }
+
+instance [AddCommGroup α] : CommGroup (Multiplicative α) :=
+  { Multiplicative.group, Multiplicative.commMonoid with }
+
+/-- Reinterpret `α →+ β` as `multiplicative α →* multiplicative β`. -/
+@[simps]
+def AddMonoidHom.toMultiplicative [AddZeroClass α] [AddZeroClass β] :
+    (α →+ β) ≃ (Multiplicative α →* Multiplicative β) where
+  toFun f := ⟨fun a => ofAdd (f (toAdd a)), f.2, f.3⟩
+  invFun f := ⟨fun a => toAdd (f (ofAdd a)), f.2, f.3⟩
+  left_inv x := by
+    ext
+    rfl
+  right_inv x := by
+    ext
+    rfl
+#align add_monoid_hom.to_multiplicative AddMonoidHom.toMultiplicative
+
+/-- Reinterpret `α →* β` as `additive α →+ additive β`. -/
+@[simps]
+def MonoidHom.toAdditive [MulOneClass α] [MulOneClass β] :
+    (α →* β) ≃ (Additive α →+ Additive β) where
+  toFun f := ⟨fun a => ofMul (f (toMul a)), f.2, f.3⟩
```

Comment:
TODO: this is currently erroring. It seems the `CoeFun` instance search for the bundled morphism fails somehow.